script.js

```diff
@@ -0,0 +1,49 @@
+const userInput = document.getElementById('user-input');
+const checkBtn = document.getElementById('check-btn');
+const clearBtn = document.getElementById('clear-btn');
+const resultsDiv = document.getElementById('results-div');
+
+const checkValidNumber = (input) => {
+  if (input === '') {
+    alert('Please provide a phone number');
+    return;
+  }
+  const countryCode = '^(1\\s?)?';
+  const areaCode = '(\\([0-9]{3}\\)|[0-9]{3})';
+  const spacesDashes = '[\\s\\-]?';
+  const phoneNumber = '[0-9]{3}[\\s\\-]?[0-9]{4}$';
+  const phoneRegex = new RegExp(
+    `${countryCode}${areaCode}${spacesDashes}${phoneNumber}`,
+  );
+
+  const pTag = document.createElement('p');
+  pTag.className = 'results-text';
+  if (phoneRegex.test(input)) {
+    pTag.style.color = '#00471b';
+  } else {
+    pTag.style.color = '#4d3800';
+  }
+
+  pTag.appendChild(
+    document.createTextNode(
+      `${phoneRegex.test(input) ? 'Valid' : 'Invalid'} US number: ${input}`,
+    ),
+  );
+  resultsDiv.appendChild(pTag);
+};
+
+checkBtn.addEventListener('click', () => {
+  checkValidNumber(userInput.value);
+  userInput.value = '';
+});
+
+userInput.addEventListener('keydown', (e) => {
+  if (e.key === 'Enter') {
+    checkValidNumber(userInput.value);
+    userInput.value = '';
+  }
+});
+
+clearBtn.addEventListener('click', () => {
+  resultsDiv.textContent = '';
+});
```

Comment:
- [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  
  // Here add all of you code that belongs to event listeners
})
```